src/Carbon/Carbon.php

```diff
@@ -1500,22 +1512,29 @@ public static function hasTestNow()
     /**
      * Determine if a time string will produce a relative date.
      *
+     * Return false when time is absolute or otherwise return array with two elements.
+     * First element is always true, second will be true if microseconds should be cleared.
+     *
      * @param string $time
      *
-     * @return bool true if time match a relative date, false if absolute or invalid time string
+     * @return false|array
```

Comment:
BC means 2.x, but 2.x will be for PHP >= 7.1.8 only on which this fix will no longer be relevant

Reply:
Could be moved to a different method.